### ISSUE TITLE
Compatibility enhancement by support of "module del"

### ISSUE THREAD

**Issue by wenzler** [cea-hpc/modules#200]
Shows how old the modules are, that we have still in use...
module3 has support for this old, undocumented(?) alias for unload 
(doesn't show anywhere in module help) 

Still need advise on how/what to gift you :>

Simple patch below
````
--- modulecmd.tcl~      2018-10-12 15:41:09.000000002 +0200
+++ modulecmd.tcl       2018-10-12 15:41:09.000000002 +0200
@@ -1349,7 +1349,7 @@
    # resolve command if alias or shortcut name used
    switch -regexp -- $command {
       {^(add|lo)}           {set command "load"}
-      {^(rm|unlo)}          {set command "unload"}
+      {^(rm|unlo|del|remove)}          {set command "unload"}
       {^(ref|rel)}          {set command "reload"}
       {^sw}                 {set command "switch"}
       {^(di|show)}          {set command "display"}
````

**Comment by xdelaruelle** [comment]
> Still need advise on how/what to gift you :>
Pull requests, bug reports and their fixes are what I am fond of ;). And on top of these kind of contributions, advocating the new versions of Modules around you would also be a very nice gift to me ^^

**Comment by wenzler** [comment]
Well well, I will just have to call your employer then.
Wonder if my french works out to that point? ;)

Regarding advocating: I changed globally to a slightly patched 4.1.4 this Saturday 
(Moving most of the error messages to debug, not bailing out on unload etc...)
as for the existing jenkins jobs are awfully picky to get any output from modulecmd

I had lots of people with modules that die in the middle and not finish up or load like they did before. 
There is lots of places where variables get updated in between module calls (reset,unset etc)
or even variables being set to the same value in different modules that causes errors like unload will not work/env corrupted as the counters are not matching...Ill have examples of that for you next week.

Just added another alias to unload:  |remove
Then threres (^(pu|clear) added to purge - its not ideal but ... people

The fault tolerance of modules 3 is pretty high :D  
but I don't dare to go to trunk.
Lets see how the day ends.
Will keep you posted

Thanks!

**Comment by xdelaruelle** [comment]
> Well well, I will just have to call your employer then.
Wonder if my french works out to that point? ;)

^^

> I had lots of people with modules that die in the middle and not finish up or load like they did before.
There is lots of places where variables get updated in between module calls (reset,unset etc)
or even variables being set to the same value in different modules that causes errors like unload will not work/env corrupted as the counters are not matching...Ill have examples of that for you next week.

Perfect. I am sure reference counter error handling could be improved.

> Just added another alias to unload: |remove

Found that when working on `del`. Fix is ongoing.

> Then threres (^(pu|clear) added to purge - its not ideal but ... people

If you want the `clear` command to be re-integrated in new version, feel free to add a ticket on that. Not so hard to re-introduce, but this is not the same meaning than `purge` (as nothing is unloaded, just the module-related variables that are cleared)
